synapse/third_party_rules/access_rules.py

```diff
@@ -80,6 +85,7 @@ def on_create_room(self, requester, config, is_requester_admin):
         """
         is_direct = config.get("is_direct")
         rules_in_initial_state = False
+        rule = ""
```

Comment:
> this seems odd. why "" rather than None, and why have it at all given its value will never be used?

It will:

https://github.com/matrix-org/synapse/blob/aeda7fde78de2ac73b146f47a29d5da779fa7e2e/synapse/third_party_rules/access_rules.py#L136

https://github.com/matrix-org/synapse/blob/aeda7fde78de2ac73b146f47a29d5da779fa7e2e/synapse/third_party_rules/access_rules.py#L144

> (a better solution might be to get rid of rules_in_initial_state and set rule = None to start with - then test if rule is None at line 115)

That sounds indeed like a good idea, thanks!